CMakeLists.txt

```diff
@@ -140,3 +139,44 @@ if(NOT TARGET uninstall)
       COMMAND ${CMAKE_COMMAND} -P
       ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
 endif()
+
+# ---[ Make configuration files for cmake to allow dependent libraries
+# easier access to Caffe2.
+
+# Specific interface link libraries for caffe2
+if ((NOT USE_GLOG) OR (NOT USE_GFLAGS) OR CAFFE2_USE_CUSTOM_PROTOBUF)
```

Comment:
Not entirely true. If CAFFE2_USE_CUSTOM_PROTOBUF then Caffe2 will link libprotobuf statically, so no need for transitive dependency of libraries. However caffe2_pb.h will include google/protobuf/... so that will be a problem unless we also install those (probably not a good idea...)

Reply:
Yeah, basically if any of these is set true, the exported config file is going to be buggy and I haven't found a good solution for that... Currently marking it as a warning.